### PR TITLE
feat!: replace submitter GUI with bundle GUI submit

### PR DESCRIPTION
**Pull request by crowecawcaw**
### What was the problem/requirement? (What/Why)
The submitter Python script currently calls a second Python application to show the submitter GUI. This setup requires a second Python installation, manually set environment variables, and extra code.

### What was the solution? (How)
Have the submitter script invoke the Deadline GUI bundle submitter instead.

### What is the impact of this change?
Less code and dependencies. Installation is far simpler - just copying a single script to Keyshot's scripts directory. No special environment variables required.

### How was this change tested?
- Added unit tests.
- Submitted jobs from Keyshot
- Verified that settings from previous submissions (parameters and attachments) are sticky and loaded back for the next submission *as long as* the submission has the same name.

### Was this change documented?
Yes.

### Is this a breaking change?
Not quite. This new submitter can work in place of the old one with no additional steps. It looks slightly different since it uses the bundle GUI submitter instead of the the custom submitter GUI we previously had. Also if job settings are sticky only if the job name stays the same (whereas previously the settings were associated with the scene file, not the submission name).

----

*By submitting this pull request, I confirm that you can use, modify, copy, and redistribute this contribution, under the terms of your choice.*

### CODE REVIEW

File: src/deadline/keyshot_submitter/Submit to AWS Deadline Cloud.py
Comment (lucaseck) [inline src/deadline/keyshot_submitter/Submit to AWS Deadline Cloud.py]: nit: We don't have a great way to ensure this gets updated right now, but can it be bumped up to 0.1.3 which is the current latest

File: src/deadline/keyshot_submitter/default_keyshot_job_template.yaml
Comment (lucaseck) [inline src/deadline/keyshot_submitter/default_keyshot_job_template.yaml:1]: Not as part of this PR, but in general I think this should be moved to https://github.com/aws-deadline/deadline-cloud-samples as an example

Comment (crowecawcaw) [reply src/deadline/keyshot_submitter/default_keyshot_job_template.yaml:1]: Makes sense. I'll leave it for the moment but can follow up.

File: README.md
Comment (lucaseck) [inline README.md]: nit: Can this be updated to reflect both local and system install locations like we are doing down below

File: install_builder/deadline-cloud-for-keyshot.xml
Comment (lucaseck) [inline install_builder/deadline-cloud-for-keyshot.xml]: `Install the integrated KeyShot submitter files to the installation directory` should be removed as well since the only file being "installed" is the `Submit to AWS Deadline Cloud` script mentioned on the line below.

File: pyproject.toml
Comment (lucaseck) [inline pyproject.toml:1]: "deadline == 0.48.*" will need to be bumped. Interesting case though where `0.48.5` is the minimum required, so we will likely need to temporarily lock to `"deadline >= 0.48.5,< 0.49"` or something similar

File: src/deadline/keyshot_submitter/Submit to AWS Deadline Cloud.py
Comment (lucaseck) [inline src/deadline/keyshot_submitter/Submit to AWS Deadline Cloud.py:24]: Normally any of the JA lists are stored as sets, is there a reason I'm missing why these need to be lists?

File: src/deadline/keyshot_submitter/Submit to AWS Deadline Cloud.py
Comment (lucaseck) [inline src/deadline/keyshot_submitter/Submit to AWS Deadline Cloud.py]: I believe `CHOOSE_OUTPUT_FILE` exists. Should that be used instead?

File: src/deadline/keyshot_submitter/Submit to AWS Deadline Cloud.py
Comment (lucaseck) [inline src/deadline/keyshot_submitter/Submit to AWS Deadline Cloud.py:250]: nit: return value typing can be expanded, looks like there are a few other places as well that I won't leave a comment on each one.

File: install_builder/deadline-cloud-for-keyshot.xml
Comment (joel-wong-aws) [reply install_builder/deadline-cloud-for-keyshot.xml]: Updated!

File: README.md
Comment (joel-wong-aws) [reply README.md]: Updated!

File: pyproject.toml
Comment (joel-wong-aws) [reply pyproject.toml:1]: Updated to `deadline >= 0.48.5`. I'm not sure why `<0.49` would be needed?

File: src/deadline/keyshot_submitter/Submit to AWS Deadline Cloud.py
Comment (joel-wong-aws) [reply src/deadline/keyshot_submitter/Submit to AWS Deadline Cloud.py:24]: These are saved as JSON, which doesn't support sets, so we use a list instead

File: src/deadline/keyshot_submitter/Submit to AWS Deadline Cloud.py
Comment (joel-wong-aws) [reply src/deadline/keyshot_submitter/Submit to AWS Deadline Cloud.py]: Yup!

File: src/deadline/keyshot_submitter/Submit to AWS Deadline Cloud.py
Comment (joel-wong-aws) [reply src/deadline/keyshot_submitter/Submit to AWS Deadline Cloud.py:250]: I don't think expanding the return type adds much value here, but I've added docstrings indicating that the return values are json-convertible!

File: pyproject.toml
Comment (lucaseck) [reply pyproject.toml:1]: Will likely have to be bumped again from the pull of 0.48.5. We normally pin to 0.x.* so I believe just pinning to >= 0.48.5 would allow 0.49 etc automatically which right now we don't want to happen, each minor version increment we want to be in control of.

File: README.md
Comment (lucaseck) [inline README.md]: Should we add a note here that deadline should be > 0.48.7 for sticky settings to work at least? The update to the pyproject.toml won't enforce anything for the submitter, only the adaptor, so it's up to the user to have a new enough version of deadline present for sticky settings to work.

Comment (joel-wong-aws) [reply README.md]: Added a note and updated the recommended `pip install` instruction!

File: README.md
Comment (lucaseck) [inline README.md]: nit: For the submitter I don't think we need to recommend pinning to < 0.49. We will have to update that every time and it's not required IMO.

File: README.md
Comment (epmog) [inline README.md]: why specify PySide6 in the command? `deadline[gui]` already brings the version of `Pyside6-Essentials` that it requires. Whereas just PySide6 will bring a lot of additional addons.

Should probably just specify installing the keyshot package to ensure we get a compatible deadline version without needing to update this all the time. Can still specify deadline[gui] to pull in that optional dep.

`pip install deadline-cloud-for-keyshot "deadline[gui]"`

should do the trick

File: README.md
Comment (joel-wong-aws) [reply README.md]: I'm confused. Didn't we want to pin this based on an [earlier discussion](https://github.com/aws-deadline/deadline-cloud-for-keyshot/pull/93#discussion_r1655374540)?

File: src/deadline/keyshot_submitter/Submit to AWS Deadline Cloud.py
Comment (epmog) [inline src/deadline/keyshot_submitter/Submit to AWS Deadline Cloud.py]: super nit: this line can be deleted.

File: src/deadline/keyshot_submitter/Submit to AWS Deadline Cloud.py
Comment (epmog) [inline src/deadline/keyshot_submitter/Submit to AWS Deadline Cloud.py]: copy/paste? Should indicate it failed to read deadline-cli output as json

File: src/deadline/keyshot_submitter/Submit to AWS Deadline Cloud.py
Comment (epmog) [inline src/deadline/keyshot_submitter/Submit to AWS Deadline Cloud.py:361]: Do we need to check if the scene has been saved first? This might not exist, or other settings may fail to be set?

File: README.md
Comment (joel-wong-aws) [reply README.md]: Removed PySide6 from the `pip install` command. Is `pip install deadline-cloud-for-keyshot` necessary? If someone is running these instructions, I understand that they would be doing some sort of local development and having an install of `deadline-cloud-for-keyshot` may cause more confusion around import/path priority? If someone doesn't want to do development of the KeyShot submitter, I understand that they would just install it from the Deadline Cloud binaries following our public docs. We could mention that here as well once we release the latest binaries.

File: src/deadline/keyshot_submitter/Submit to AWS Deadline Cloud.py
Comment (joel-wong-aws) [reply src/deadline/keyshot_submitter/Submit to AWS Deadline Cloud.py]: Removed!

File: src/deadline/keyshot_submitter/Submit to AWS Deadline Cloud.py
Comment (joel-wong-aws) [reply src/deadline/keyshot_submitter/Submit to AWS Deadline Cloud.py]: Updated!

File: src/deadline/keyshot_submitter/Submit to AWS Deadline Cloud.py
Comment (joel-wong-aws) [reply src/deadline/keyshot_submitter/Submit to AWS Deadline Cloud.py:361]: Out of scope, the existing implementation also assumes that the file exists

File: README.md
Comment (epmog) [reply README.md]: Gotcha. In my mind this was related to the manual installation instructions, rather than solely the development instructions. Looking at it right now, there's nothing to indicate that it's development, manual installation, or a continuation from the submitter installer - the section is just "Using the submitter".

Future improvement we can punt - but we'll want to make this clear.

File: src/deadline/keyshot_submitter/Submit to AWS Deadline Cloud.py
Comment (epmog) [reply src/deadline/keyshot_submitter/Submit to AWS Deadline Cloud.py:361]: 😬 fair enough - resolving.

**Review by epmog — DISMISSED** [review]

File: README.md
Comment (epmog) [reply README.md]: > Is pip install deadline-cloud-for-keyshot necessary?

It depends on what the instructions are targeting. Doing this gets you the correct version of deadline which will then mean we don't have to keep updating these instructions on new releases.


1. If you're doing a local build, `pip install`ing the built wheel will grab you the correct version.
2. If you're doing a manual install, pip installing from pypi gets you the right dependency versions for the installed version
3. If you're continuing from a submitter installer build none of this makes sense.

File: README.md
Comment (lucaseck) [reply README.md]: Spoke offline, and this can be punted until after merging. No sense holding this PR up more.

**Review by lucaseck — DISMISSED** [review]

**Review by lucaseck — APPROVED** [review]

**Review by epmog — APPROVED** [review]

**Pull request merged**
